### PR TITLE
[gardening] Fix shellcheck warnings in recently introduced file find-overlay-deps-closure.sh

### PR DESCRIPTION
**Pull request by practicalswift**
Ping @erg - please review :-)

### CODE REVIEW

**Comment by practicalswift** [comment]
@swift-ci please smoke test

File: utils/find-overlay-deps-closure.sh [hunk 1]
```diff
@@ -39,9 +41,10 @@ function find_deps() {
 
     local REGEX="./Frameworks/(${OVERLAY_NAME_ALTERNATION}).framework/.*|.*/usr/include/(xpc|dispatch|os|objc|simd)/.*\.h"
 
-    IFS='\
+    IFS='
```
Comment (erg) [inline utils/find-overlay-deps-closure.sh]: This changes the behavior and makes it take twice as long.

Put this inside the line 48 loop and see how it calls the regex on ``\`` lines as well. 
```            echo "line: $line" >&2  # line 49```

**Comment by erg** [comment]
Looks good if you fix the speed regression. Thanks!

**Comment by practicalswift** [comment]
@erg Fixed! :-)

**Comment by practicalswift** [comment]
@swift-ci please smoke test and merge

**Comment by practicalswift** [comment]
@swift-ci please test

**Pull request merged**